CameraXBasic/app/src/main/java/com/android/example/cameraxbasic/fragments/GalleryFragment.kt

```diff
@@ -127,12 +134,14 @@ class GalleryFragment internal constructor() : Fragment() {
 
         // Handle delete button press
         view.findViewById<ImageButton>(R.id.delete_button).setOnClickListener {
-            AlertDialog.Builder(view.context, android.R.style.Theme_Material_Dialog)
-                    .setTitle(getString(R.string.delete_title))
-                    .setMessage(getString(R.string.delete_dialog))
-                    .setIcon(android.R.drawable.ic_dialog_alert)
-                    .setPositiveButton(android.R.string.yes) { _, _ ->
-                        mediaList.getOrNull(mediaViewPager.currentItem)?.let { mediaFile ->
+
+            mediaList.getOrNull(mediaViewPager.currentItem)?.let { mediaFile ->
```

Comment:
This logic does not seem quite right. If we are going to check whether `mediaList[mediaViewPager.currentItem]` is present then it's better to do it when the button is clicked and not when the button listener is created. I would revert this logic to the way it was before.

Reply:
done